api/src/main/java/com/github/commonsrdf/api/BlankNode.java

```diff
@@ -43,77 +57,55 @@
 	/**
 	 * Return a <a href=
 	 * "http://www.w3.org/TR/rdf11-concepts/#dfn-blank-node-identifier"
-	 * >label</a> for the blank node. This is not a serialization/syntax label.
-	 * It should be uniquely identifying within the local scope it is created in
-	 * but has no uniqueness guarantees other than that.
-	 * <p>
-	 * In particular, the existence of two objects of type {@link BlankNode}
-	 * with the same value returned from {@link #internalIdentifier()} are not
-	 * equivalent unless they are known to have been created in the same local
-	 * scope (see {@link #equals(Object)})
+	 * >identifier</a> for the blank node as an {@link UUID}. 
+	 * This is not a serialization/syntax label.
+	 * It uniquely identify the blank node, but does not
+	 * come with any guarantees for persistence or resolution.
 	 * <p>
-	 * An example of a local scope may be an instance of a Java Virtual Machine
-	 * (JVM). In the context of a JVM instance, an implementor may support
-	 * insertion and removal of {@link Triple} objects containing Blank Nodes
-	 * without modifying the blank node labels.
+	 * Two objects of the type <code>BlankNode</code> with the same
+	 * value returned from {@link #identifier()} are equivalent, 
+	 * and thus are in the same local scope.
+	 * (see {@link #equals(Object)}).
 	 * <p>
-	 * Another example of a local scope may be a <a
-	 * href="http://www.w3.org/TR/rdf11-concepts/#section-rdf-graph">Graph</a>
-	 * or <a
-	 * href="http://www.w3.org/TR/rdf11-concepts/#section-dataset">Dataset</a>
-	 * created from a single document. In this context, an implementor should
-	 * reasonably guarantee that the label returned by getLabel only maps to
-	 * equivalent blank nodes in the same Graph or Dataset, but they may not
-	 * guarantee that it is unique for the JVM instance. In this case, the
-	 * implementor may support a mechanism to provide a mapping for blank nodes
-	 * between Graph or Dataset instances to guarantee their uniqueness.
+	 * On the other hand, two <code>BlankNode</code> objects
+	 * with different <code>identifier()</code> values are not 
+	 * necessarily different.
 	 * <p>
-	 * If implementors support <a
-	 * href="http://www.w3.org/TR/rdf11-concepts/#section-skolemization"
-	 * >Skolemisation</a>, they may map instances of {@link BlankNode} objects
-	 * to {@link IRI} objects to reduce scoping issues.
-	 * <p>
-	 * It is not a requirement for the internal identifier to be a part of the
-	 * {@link #ntriplesString()}, except that two BlankNode instances with the
-	 * same internalIdentifier() and same local scope should have the same
-	 * {@link #ntriplesString()}.
+	 * It is not a requirement for this UUID <code>identifier</code> 
+	 * to form part of the {@link #ntriplesString()}. 
+	 *
+	 * @see #equals(Object)
 	 *
-	 * @return An internal, system identifier for the {@link BlankNode}.
+	 * @return An unique identifier for the {@link BlankNode}.
 	 */
-	String internalIdentifier();
+	UUID identifier();
```

Comment:
On 06/02/15 01:07, Peter Ansell wrote:
> In api/src/main/java/com/github/commonsrdf/api/BlankNode.java
> <https://github.com/commons-rdf/commons-rdf/pull/63#discussion_r24215817>:
>
>>  	 */
>> -	String internalIdentifier();
>> +	UUID identifier();
>
> It may be simpler to return a consistent String representation of the
> UUID, rather than mandating it to be an actual UUID object.

I agree, and we should not pick one unique identifier mechanism when 
there are alternatives.

1. skolemization as described in RDF 1.1
2. UUID (not always small device friendly)
3. If a system is only for a specific use, "unique" maybe more limited 
range.

Having just `internalIdentifier` keeps that flexibility.